src/System.Memory/Directory.Build.props

```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <!-- System.Memory has forwarded types into the runtime on netcoreapp/uap
          It must win over assemblies versioned at 4.0.* -->
-    <AssemblyVersion>4.1.0.0</AssemblyVersion>
+    <AssemblyVersion>4.2.0.0</AssemblyVersion>
```

Comment:
Because we changed the surface area? @ericstj?

Reply:
Technically we don't have to change the assembly version since it's only shipping in netcoreapp, though given the package is currently disabled there isn't anything "wrong" with changing the version and it can act as a safeguard should someone try to build against the new surface and run on old frameworks.